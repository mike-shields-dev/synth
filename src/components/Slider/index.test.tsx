import { render, screen } from "@testing-library/react";
import { Slider } from ".";

const validProps = {
    groupName: "groupName",
    parameter: "parameter",
    displayName: "displayName",
}

describe("Slider", () => {
    it('matches snapshot', () => {
        const { container } = render(<Slider {...validProps} />);

        expect(container.firstChild).toMatchSnapshot();
    });

    beforeEach(() => render(< Slider {...validProps} />));

    it('renders a Slider component', () => {
        expect(screen.getByTestId("Slider")).toBeInTheDocument();
    });

    it('renders a single input with the type "range"', () => {
        expect(screen.getAllByRole("slider")).toHaveLength(1);
    });

    it('renders an associated input label with the given displayName', () => {
        expect(screen.getByRole("slider", { name: validProps.displayName })).toBeInTheDocument();
    });
});