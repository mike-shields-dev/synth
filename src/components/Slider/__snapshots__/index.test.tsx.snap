// Vitest Snapshot v1

exports[`Slider > matches snapshot 1`] = `
<div
  data-testid="Slider"
>
  <label
    for="groupName:parameter"
  >
    displayName
  </label>
  <input
    id="groupName:parameter"
    type="range"
  />
</div>
`;
